huaweicloud/services/live/resource_huaweicloud_live_domain.go

```diff
@@ -0,0 +1,325 @@
+
+			"ingest_domain_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"status": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{domainStatusOn, domainStatusOff}, false),
+				Computed:     true,
+			},
+
+			"cname": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceDomainCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	c := meta.(*config.Config)
+	region := c.GetRegion(d)
+	client, err := c.HcLiveV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating Live v1 client: %s", err)
+	}
+
+	createOpts, err := buildCreateParams(d, region)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	log.Printf("[DEBUG] Create Live domain params : %#v", createOpts)
+
+	_, err = client.CreateDomain(createOpts)
+	if err != nil {
+		return diag.Errorf("error creating Live domain: %s", err)
+	}
+
+	d.SetId(createOpts.Body.Domain)
+
+	err = waitingForDomainStatus(ctx, client, d.Id(), model.GetDecoupledLiveDomainInfoStatusEnum().ON,
+		d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	// associate the streaming domain name with an ingest domain
+	err = associatingDomain(d, client)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return resourceDomainRead(ctx, d, meta)
+}
+
+func resourceDomainRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	c := meta.(*config.Config)
+	region := c.GetRegion(d)
+	client, err := c.HcLiveV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating Live v1 client: %s", err)
+	}
+
+	domain := d.Id()
+	response, err := client.ShowDomain(&model.ShowDomainRequest{Domain: &domain})
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving Live domain")
+	}
+
+	r := *response.DomainInfo
+	detail := r[0]
+
+	mErr := multierror.Append(
+		d.Set("region", region),
+		d.Set("name", detail.Domain),
+		d.Set("type", utils.MarshalValue(detail.DomainType)),
+		d.Set("status", utils.MarshalValue(detail.Status)),
+		d.Set("ingest_domain_name", detail.RelatedDomain),
+		d.Set("cname", detail.DomainCname),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func resourceDomainUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	c := meta.(*config.Config)
+	region := c.GetRegion(d)
+	client, err := c.HcLiveV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating Live v1 client: %s", err)
+	}
+
+	domainName := d.Get("name").(string)
+
+	// associate the streaming domain name with an ingest domain Or delete association
+	if d.HasChange("ingest_domain_name") {
+		ingetstDomainNameOld, ingetstDomainName := d.GetChange("ingest_domain_name")
+
+		if ingetstDomainName == "" {
+			err = deleteAssociation(client, domainName, ingetstDomainNameOld.(string))
+		} else {
+			err = associatingDomain(d, client)
```

Comment:
can we associate a new domain without disassociate the old one?

Reply:
yes, of course